### DISCUSSION TITLE
ws.js:90 WebSocket connection to 'wss://relay.walletconnect.com/?auth=...' failed: Invalid frame header

### DISCUSSION THREAD

**Discussion by Svg70** [WalletConnect/walletconnect-monorepo#3310] [Developer Support]
The specific error message I encountered is as follows:
"ws.js:90 WebSocket connection to 'wss://relay.walletconnect.com/?auth=...' failed: Invalid frame header."

It is important to note that this error occurs consistently on my Mac, making it impossible for me to utilize WalletConnect. However, I have observed that on Windows, the service works perfectly fine without any issues.

Given the situation, I am eager to understand the possible reasons behind this discrepancy. Could you kindly provide guidance on what might be causing the WebSocket connection failure on my Mac? Additionally, I would appreciate any instructions or steps to resolve this issue, allowing me to use WalletConnect seamlessly on my Mac as well.

**Comment by blabno** [comment]
Same thing here. On linux I'm getting "Invalid frame header" but on windows, stuff works.
On Linux I've tried brave, chrome and firefox, all fail. On firefox the error looks like this 
> The connection `wss://relay.walletconnect.com?...` was interrupted while the page was loading.
The only difference in urls between linux and windows I see (apart from obvious auth query param) is `ua` query parameter.
Values of `ua`:
* Linux: wc-2/js-2.10.1/linux-chrome-120.0.0/browser:app.uniswap.org
* Windows:  wc-2/js-2.7.0/windows10-chrome-120.0.0/browser:app.squidrouter.com

**Reply by blabno** [reply]
Nevermind, my clock was few minutes in the future. Syncing it with ntp-client resolved the issue.
I guess the server was closing connection because JWT in auth query param had issue date in future.
